sys/net/gnrc/netif/gnrc_netif.c

```diff
@@ -1347,20 +1347,84 @@ void gnrc_netif_default_init(gnrc_netif_t *netif)
 #endif
 }
 
+#ifdef MODULE_GNRC_NETIF_EVENTS
+/**
+ * @brief   Call the ISR handler from an event
+ *
+ * @param[in]   evp     pointer to the event
+ */
+static void _event_handler_isr(event_t *evp)
+{
+    gnrc_netif_t *netif = container_of(evp, gnrc_netif_t, event_isr);
+    netif->dev->driver->isr(netif->dev);
+}
+#endif /* MODULE_GNRC_NETIF_EVENTS */
+
+/**
+ * @brief   Process any pending events and wait for IPC messages
+ *
+ * This function will block until an IPC message is received. Events posted to
+ * the event queue will be processed while waiting for messages.
+ *
+ * @param[in]   netif   gnrc_netif instance to operate on
+ * @param[out]  msg     pointer to message buffer to write the first received message
+ *
+ * @return >0 if msg contains a new message
+ */
+static void _process_events_await_msg(gnrc_netif_t *netif, msg_t *msg)
+{
+#ifdef MODULE_GNRC_NETIF_EVENTS
```

Comment:
I suggest to change this to a regular if-else chain (with IS_ACTIVE). Then we get compile checks for free

Reply:
That's not possible, some members of `gnrc_netif_t` are only available when this module is enabled. Modifying this to `IS_ACTIVE` will result in compilation failures when `GNRC_NETIF_EVENTS` is not enabled.

Reply:
you can simply define a static inline function:
```c
static inline _get_evq(netif_t *netif)
{
#if IS_ACTIVE(MODULE_GNRC_NETIF_EVENTS)
    return netif->evq;
#else 
    return NULL;
#endif 
}
```

Then you call that function in the while loop.

Reply:
This function does not compile when `evq` is not a member of `gnrc_netif_t`, which is the case when `MODULE_GNRC_NETIF_EVENTS` because of the `ifdef` guards [here](https://github.com/RIOT-OS/RIOT/pull/13669/files#diff-36d38c9fc81553f08febd7c298fd7168R100)

Reply:
> This function does not compile when `evq` is not a member of `gnrc_netif_t`, which is the case when `MODULE_GNRC_NETIF_EVENTS` because of the `ifdef` guards [here](https://github.com/RIOT-OS/RIOT/pull/13669/files#diff-36d38c9fc81553f08febd7c298fd7168R100)

Then it should still work with @jia200x's last proposal.

Reply:
> Then it should still work with @jia200x's last proposal.

Oh, of course, I misread his last suggestion.

Reply:
Okay, while this is possible, it is starting to become more and more messy. For the solution posted by @jia200x I have to include both `event.h` and `thread_flags.h`. For `thread_flags.h` to be included it is required to enable `MODULE_CORE_THREAD_FLAGS` increasing the build size whether we're using `MODULE_GNRC_NETIF_EVENTS` or not. 

Conditionally including `event.h` and `thread_flags.h` doesn't work because the code between the regular if-else chain (with IS_ACTIVE) requires those for one of the branches to compile.

Am I missing something silly here that allows me to get this to work or is the current `#ifdef` really the only way out here? I honestly don't want to start guarding every individual troublesome call in this snippet if we can avoid it.

Reply:
It should be `IS_USED()` btw not, `IS_ACTIVE()`. If you use `#if` (not `if ()` it _should_ work even if you `#ifdef` the calls above. Let me give you some suggestions.

Reply:
The root of the problem here is the #error in thread_flags header. It's misplaced IMO. It should be in the C file that defines the thread flags function.

I can open a PR to move it (unless you are faster :), it should be a low hanging fruit.

Reply:
It's there: https://github.com/RIOT-OS/RIOT/pull/13671

And @miri64 is right. It should be IS_USED for modules